src/libcrun/handlers/wasmtime.c

```diff
@@ -0,0 +1,233 @@
+  wasmtime_linker_t *(*wasmtime_linker_new) (wasm_engine_t * engine);
+  wasmtime_error_t *(*wasmtime_linker_define_wasi) (wasmtime_linker_t * linker);
+  wasmtime_error_t *(*wasmtime_module_new) (
+      wasm_engine_t * engine,
+      const uint8_t *wasm,
+      size_t wasm_len,
+      wasmtime_module_t **ret);
+  void (*wasi_config_inherit_argv) (wasi_config_t * config);
+  void (*wasi_config_inherit_env) (wasi_config_t * config);
+  void (*wasi_config_inherit_stdin) (wasi_config_t * config);
+  void (*wasi_config_inherit_stdout) (wasi_config_t * config);
+  void (*wasi_config_inherit_stderr) (wasi_config_t * config);
+  wasmtime_error_t *(*wasmtime_context_set_wasi) (wasmtime_context_t * context, wasi_config_t * wasi);
+  wasmtime_error_t *(*wasmtime_linker_module) (
+      wasmtime_linker_t * linker,
+      wasmtime_context_t * store,
+      const char *name,
+      size_t name_len,
+      const wasmtime_module_t *module);
+  wasmtime_error_t *(*wasmtime_linker_get_default) (
+      const wasmtime_linker_t *linker,
+      wasmtime_context_t *store,
+      const char *name,
+      size_t name_len,
+      wasmtime_func_t *func);
+  wasmtime_error_t *(*wasmtime_func_call) (
+      wasmtime_context_t * store,
+      const wasmtime_func_t *func,
+      const wasmtime_val_t *args,
+      size_t nargs,
+      wasmtime_val_t *results,
+      size_t nresults,
+      wasm_trap_t **trap);
+  void (*wasmtime_module_delete) (wasmtime_module_t * m);
+  void (*wasmtime_store_delete) (wasmtime_store_t * store);
+
+  wasm_engine_new = dlsym (cookie, "wasm_engine_new");
+  wasm_engine_delete = dlsym (cookie, "wasm_engine_delete");
+  wasm_byte_vec_delete = dlsym (cookie, "wasm_byte_vec_delete");
+  wasm_byte_vec_new_uninitialized = dlsym (cookie, "wasm_byte_vec_new_uninitialized");
+  wasi_config_new = dlsym (cookie, "wasi_config_new");
+  wasmtime_store_new = dlsym (cookie, "wasmtime_store_new");
+  wasmtime_store_context = dlsym (cookie, "wasmtime_store_context");
+  wasmtime_linker_new = dlsym (cookie, "wasmtime_linker_new");
+  wasmtime_linker_define_wasi = dlsym (cookie, "wasmtime_linker_define_wasi");
+  wasmtime_module_new = dlsym (cookie, "wasmtime_module_new");
+  wasi_config_inherit_argv = dlsym (cookie, "wasi_config_inherit_argv");
+  wasi_config_inherit_stdout = dlsym (cookie, "wasi_config_inherit_stdout");
+  wasi_config_inherit_stdin = dlsym (cookie, "wasi_config_inherit_stdin");
+  wasi_config_inherit_stderr = dlsym (cookie, "wasi_config_inherit_stderr");
+  wasi_config_inherit_env = dlsym (cookie, "wasi_config_inherit_env");
+  wasmtime_context_set_wasi = dlsym (cookie, "wasmtime_context_set_wasi");
+  wasmtime_linker_module = dlsym (cookie, "wasmtime_linker_module");
+  wasmtime_linker_get_default = dlsym (cookie, "wasmtime_linker_get_default");
+  wasmtime_func_call = dlsym (cookie, "wasmtime_func_call");
+  wasmtime_module_delete = dlsym (cookie, "wasmtime_module_delete");
+  wasmtime_store_delete = dlsym (cookie, "wasmtime_store_delete)");
```

Comment:
TODO: verify `dlsym` return values.

Reply:
I think the ')' should be dropped from the fn name

Reply:
Thanks fixed it, will undraft as soon as i test this wiring of `dlsym`